contrib/mesos/pkg/offers/offers.go

```diff
@@ -441,7 +441,7 @@ func (s *offerStorage) ageOffers() {
 }
 
 func (s *offerStorage) nextListener() *offerListener {
-	obj := s.listeners.Pop()
+	obj := s.listeners.Pop(nil)
```

Comment:
It's clear that I'm out-voted. So I'm feeling more willing to change it back. **But** consider this: the semantics are exactly the same -- there is no need for two funcs that do the exact, I repeat the **exact** same thing. `Pop(<-chan cancel)` will block until either an item is available or `cancel` is closed; if `cancel` is `nil` then it blocks forever. This is how channels work in Go. There is no mystery here: reading from a `nil` will block forever. There is no other semantic difference between the two proposed funcs - they both block until some condition arises. And in this case it's a parameterized condition that's quite clear IMHO. It's an input-only chan so it can only be read from; not written-to or closed by the `Pop` func. Either someone else closes it, or it's `nil`.

In API design there's something to be said for conciseness. The aforementioned quote seems targeted at languages that allow method overloading by permitting either (a) multiple methods with the same name, and/or; (b) methods with default args. The "matching by name" argument doesn't seem to apply to this case at all.

Would it help to create a helper var in package `queue` so that `Pop(nil)` can be written as `Pop(queue.WithoutCancel)`? Would that make things more clear?